radix-engine-tests/tests/typed_substate_layout.rs

```diff
@@ -95,6 +143,101 @@ fn test_all_scenario_commit_receipts_should_have_substate_changes_which_can_be_t
     }
 }
 
+#[test]
+fn test_all_scenario_commit_receipts_should_have_events_that_can_be_typed() {
+    let network = NetworkDefinition::simulator();
+    let mut test_runner = TestRunner::builder().build();
+
+    let mut next_nonce: u32 = 0;
+    for scenario_builder in get_builder_for_every_scenario() {
+        let epoch = test_runner.get_current_epoch();
+        let mut scenario = scenario_builder(ScenarioCore::new(network.clone(), epoch, next_nonce));
+        let mut previous = None;
+        loop {
+            let next = scenario
+                .next(previous.as_ref())
+                .map_err(|err| err.into_full(&scenario))
+                .unwrap();
+            match next {
+                NextAction::Transaction(next) => {
+                    let receipt =
+                        test_runner.execute_raw_transaction(&network, &next.raw_transaction);
+                    match &receipt.transaction_result {
+                        TransactionResult::Commit(commit_result) => {
+                            assert_receipt_events_can_be_typed(commit_result);
+                        }
+                        // Ignore other results - which may be valid in the context of the scenario
+                        _ => {}
+                    }
+                    previous = Some(receipt);
+                }
+                NextAction::Completed(end_state) => {
+                    next_nonce = end_state.next_unused_nonce;
+                    break;
+                }
+            }
+        }
+    }
+}
+
+/// We need to ensure that all of the events registered to native events are included in the typed
+/// native event model. This test checks that the events in `typed_native_events.rs` module all
+/// exist in the blueprint schema.
+#[test]
+fn typed_native_event_type_contains_all_native_events() {
+    // Arrange
+    let package_name_definition_mapping = hashmap! {
+        "ConsensusManager" => CONSENSUS_MANAGER_PACKAGE_DEFINITION.deref(),
+        "Account" => ACCOUNT_PACKAGE_DEFINITION.deref(),
+        "Identity" => IDENTITY_PACKAGE_DEFINITION.deref(),
+        "AccessController" => ACCESS_CONTROLLER_PACKAGE_DEFINITION.deref(),
+        "Pool" => POOL_PACKAGE_DEFINITION.deref(),
+        "TransactionTracker" => TRANSACTION_TRACKER_PACKAGE_DEFINITION.deref(),
+        "Resource" => RESOURCE_PACKAGE_DEFINITION.deref(),
+        "Package" => PACKAGE_PACKAGE_DEFINITION.deref(),
+        "TransactionProcessor" => TRANSACTION_PROCESSOR_PACKAGE_DEFINITION.deref(),
+        "Metadata" => METADATA_PACKAGE_DEFINITION.deref(),
+        "Royalty" => ROYALTY_PACKAGE_DEFINITION.deref(),
+        "AccessRules" => ACCESS_RULES_PACKAGE_DEFINITION.deref(),
+    };
+
+    // Act
+    let registered_events = TypedNativeEvent::registered_events();
+
+    // Assert
+    for (package_name, package_blueprints) in registered_events.into_iter() {
```

Comment:
True, seems like we can either assume that `registered_events` is comprehensive or that `package_name_definition_mapping` is comprehensive. Since we don't add an entirely new native package all that often, I think that it might be fine to manually check that `registered_events` contains all of the packages.